src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * @file src/Modules/Settings/Uexport.php
+ */
+
+namespace Friendica\Module\Settings;
+
+use Friendica\App;
+use Friendica\App\Arguments;
+use Friendica\BaseModule;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Module\BaseSettingsModule;
+
+/**
+ * Module to export user data
+ **/
+class Uexport extends BaseSettingsModule
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
+			switch ($action) {
+				case "backup":
+					SELF::uexport_all($a);
+					exit();
+					break;
+				case "account":
+					SELF::uexport_account($a);
+					exit();
+					break;
+				default:
+					exit();
+			}
+		}
+
+		/**
+		 * options shown on "Export personal data" page
+		 * list of array( 'link url', 'link text', 'help text' )
+		 */
+		$options = [
+			['settings/uexport/account', L10n::t('Export account'), L10n::t('Export your account info and contacts. Use this to make a backup of your account and/or to move it to another server.')],
+			['settings/uexport/backup', L10n::t('Export all'), L10n::t("Export your accout info, contacts and all your items as json. Could be a very big file, and could take a lot of time. Use this to make a full backup of your account \x28photos are not exported\x29")],
+		];
+		Hook::callAll('uexport_options', $options);
+
+		$tpl = Renderer::getMarkupTemplate("uexport.tpl");
+		return Renderer::replaceMacros($tpl, [
+			'$title' => L10n::t('Export personal data'),
+			'$options' => $options
+		]);
+	}
+	private function uexport_multirow($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
+		if (DBA::isResult($r)) {
+			foreach ($r as $rr) {
+				$p = [];
+				foreach ($rr as $k => $v) {
+					switch ($dbStructure[$table]['fields'][$k]['type']) {
+						case 'datetime':
+							$p[$k] = $v ?? DBA::NULL_DATETIME;
+							break;
+						default:
+							$p[$k] = $v;
+							break;
+					}
+				}
+				$result[] = $p;
+			}
+		}
+		return $result;
+	}
+
+	private function uexport_row($query) {
+		global $dbStructure;
```

Comment:
This global variable doesn't exist anymore.
```suggestion
		$dbStructure = Friendica\Database\DBStructure::definition(self::getApp()->getBasePath(), false);
```

Reply:
Why is it then even used in the original module?

Reply:
I feel like it's a forgotten reference to a global variable that used to exist and we didn't check for all the usages throughout the codebase. It shouldn't prevent the module from running, but it should output a bunch of Notice messages in your error log file.